src/vsr/replica.zig

```diff
@@ -704,11 +715,6 @@ pub fn ReplicaType(
                 return;
             }
 
-            if (self.journal.status != .recovered) {
-                log.debug("{}: on_message: waiting for journal to recover", .{self.replica});
-                return;
-            }
-
             assert(message.header.replica < self.replica_count);
```

Comment:
Can this code-path safely be entered given it may not be `.recovered` yet?

Reply:
It is `recovered`, but we don't need to test it — we don't handle incoming messages until after the replica is opened.

Reply:
(I'll add an assertion to that effect).